### DISCUSSION TITLE
How to use xdist with a custom test collector ?

### DISCUSSION THREAD

**Discussion by benoit-cty** [pytest-dev/pytest-xdist#806] [General]
Hello,

We would like to use XDist to speed up our CI but it doesn't work with our custom plugin.

Our Plugin (https://github.com/openfisca/openfisca-core/blob/master/openfisca_core/tools/test_runner.py) collect YAML file describing input of a test and expected output then run the test.

But when we activate XDist, it override our collector and collect .py file and not yaml.

Do you know if there is a way to make our plugin work with XDist ?

We have an Issue on our side with more details if needed : https://github.com/openfisca/openfisca-core/issues/1025

**Comment by RonnyPfannschmidt** [comment]
at first glance it seems like your setup never tries to configure the subprocesses to use your own collector as well

pytest-xdist starts **new** pytest sessions in the subprocesses that have never seen your plugin object

the easiest way to enable your plugin there would be to have way to use the entrypoint mechanism

else some digging is needed to make this possible

**Comment by bonjourmauko** [comment]
@RonnyPfannschmidt what I tried to do was to use the `pytest_configure_node` hook to register the plugin for each node, but apparently it is already done, and nonetheless `pytest_collect_item` does not fire up. I understand that the nodes create the subprocesses after that hook is called, but I couldn't find a way to configure the subprocesses themselves. Maybe there is a hook for that?

**Comment by RonnyPfannschmidt** [comment]
Pytest-configure in a subprocess is it, configure node cannot add plugins
